### PR TITLE
Revert potentially dangerous optimization

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

As Juani pointed out, caching the `bufferWrappedSurplus` might be dangerous, given that there are external calls between when it's initialized and the last time it's used.

I tried a few variants that preserved some of it (e.g., just reloading as necessary), but as the point of this was to save bytecode and get the Vault to fit, simply reverting that change turned out to be best. (The variants were actually slightly more bytecode.)

Luckily, due to the other changes, it still fits with this change reverted.

## Type of change

- [X?] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [ ] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [X] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: pkg/vault/contracts/Vault.sol [hunk 1]
```diff
@@ -1179,17 +1176,16 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
                 amountInUnderlying = vaultUnderlyingDelta - bufferUnderlyingSurplus;
                 // Since bufferUnderlyingSurplus was wrapped, the final amountOut needs to discount the wrapped amount
                 // that will stay in the buffer
-                amountOutWrapped = vaultWrappedDelta - bufferWrappedSurplus;
+                amountOutWrapped = vaultWrappedDelta - wrappedToken.convertToShares(bufferUnderlyingSurplus);
```
Comment (jubeira) [inline pkg/vault/contracts/Vault.sol:1179]: Maybe make `bufferWrappedSurplus` local
```suggestion
                uint256 bufferWrappedSurplus = wrappedToken.convertToShares(bufferUnderlyingSurplus)
                amountOutWrapped = vaultWrappedDelta - bufferWrappedSurplus;
```

File: pkg/vault/contracts/Vault.sol [hunk 2]
```diff
@@ -1179,17 +1176,16 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
...
                 // In a wrap operation, the underlying balance of the buffer will decrease and the wrapped balance will
                 // increase. To decrease underlying balance, we get the delta amount that was deposited
                 // (vaultUnderlyingDelta) and discounts the amount needed in the wrapping operation
                 // (amountInUnderlying). Same logic applies to wrapped balances.
                 // Note: bufferUnderlyingSurplus = vaultUnderlyingDelta - amountInUnderlying
-                //       bufferWrappedSurplus = vaultWrappedDelta - amountOutWrapped
                 bufferBalances = PackedTokenBalance.toPackedBalance(
                     bufferBalances.getBalanceRaw() - bufferUnderlyingSurplus,
-                    bufferBalances.getBalanceDerived() + bufferWrappedSurplus
+                    bufferBalances.getBalanceDerived() + (vaultWrappedDelta - amountOutWrapped)
```
Comment (jubeira) [inline pkg/vault/contracts/Vault.sol:1188]: And reuse here to save an addition:
```suggestion
                    bufferBalances.getBalanceDerived() + bufferWrappedSurplus
```

**Review by jubeira — COMMENTED** [review]
Thanks @EndymionJkb !

I have one suggestion that may save some bytecode following the spirit of the original change, but without caching the results.

Note that we also need to apply these changes to `unwrap`. I'll do one more pass after that's done.

File: pkg/vault/contracts/Vault.sol (hunk 1 above)
Comment (EndymionJkb) [reply pkg/vault/contracts/Vault.sol:1179]: I swear I tried this, and the extra assignment canceled out the benefit, so I just left it. Now it seems to save 3 bytes, though. Well, it's clearer anyway.

**Review by jubeira — COMMENTED** [review]
Looks good now!

One final ask: can we make the code symmetrical? Either option works (make the surplus variable as local as possible as `wrap`, or declare above and reuse as with `unwrap`).

Keeping the exact same structure helps readers follow through better. Same that happens with add / remove.

Sorry for being so annoying :D

**Comment by EndymionJkb** [comment]
> Looks good now!
> 
> One final ask: can we make the code symmetrical? Either option works (make the surplus variable as local as possible as `wrap`, or declare above and reuse as with `unwrap`).
> 
> Keeping the exact same structure helps readers follow through better. Same that happens with add / remove.
> 
> Sorry for being so annoying :D

NP; will do

**Pull request merged**